### ISSUE TITLE
error installing carta-beta on MacBook pro M1 chip macOS Monterey 12.3.1

### ISSUE THREAD

**Issue by jehibbard** [CARTAvis/homebrew-tap#12]
Hi. I got a new MBP and tried to install CARTA beta on it using home-brew, but got the following error message:

------------
% brew install cartavis/tap/carta-beta
==> Downloading https://registry.npmjs.org/carta-frontend/-/carta-frontend-3.0.0-beta.2b.tgz
Already downloaded: /Users/jhibbard/Library/Caches/Homebrew/downloads/3ffe2c7345a8ebbfd0cb132fdc5a5566f73496de6489e677768bb5b8e43b925e--carta-frontend-3.0.0-beta.2b.tgz
==> Cloning https://github.com/CARTAvis/carta-backend.git
Updating /Users/jhibbard/Library/Caches/Homebrew/carta-beta--git
==> Checking out tag v3.0.0-beta.2b
HEAD is now at a544511 Bumped tag version.
HEAD is now at a544511 Bumped tag version.
Entering 'carta-protobuf'
Entering 'carta-scripting-grpc'
Entering 'third-party/cxxopts'
Entering 'third-party/image-generator'
Entering 'third-party/json'
Entering 'third-party/spdlog'
Entering 'third-party/sse2neon'
Entering 'third-party/uWebSockets'
Synchronizing submodule url for 'fuzzing/libEpollFuzzer'
Synchronizing submodule url for 'libdeflate'
Synchronizing submodule url for 'uSockets'
Entering 'third-party/uWebSockets/fuzzing/libEpollFuzzer'
Entering 'third-party/uWebSockets/libdeflate'
Entering 'third-party/uWebSockets/uSockets'
/Users/jhibbard/Library/Caches/Homebrew/carta-beta--git/carta-protobuf
/Users/jhibbard/Library/Caches/Homebrew/carta-beta--git/carta-scripting-grpc
/Users/jhibbard/Library/Caches/Homebrew/carta-beta--git/third-party/cxxopts
/Users/jhibbard/Library/Caches/Homebrew/carta-beta--git/third-party/image-generator
/Users/jhibbard/Library/Caches/Homebrew/carta-beta--git/third-party/json
/Users/jhibbard/Library/Caches/Homebrew/carta-beta--git/third-party/spdlog
/Users/jhibbard/Library/Caches/Homebrew/carta-beta--git/third-party/sse2neon
/Users/jhibbard/Library/Caches/Homebrew/carta-beta--git/third-party/uWebSockets
/Users/jhibbard/Library/Caches/Homebrew/carta-beta--git/third-party/uWebSockets/fuzzing/libEpollFuzzer
/Users/jhibbard/Library/Caches/Homebrew/carta-beta--git/third-party/uWebSockets/libdeflate
/Users/jhibbard/Library/Caches/Homebrew/carta-beta--git/third-party/uWebSockets/uSockets
==> Installing carta-beta from cartavis/tap
==> git submodule update --recursive --init
==> cmake .. -DCMAKE_PREFIX_PATH=/opt/homebrew/Cellar/carta-beta/3.0.0-beta/lib -DCMAKE_INCLUDE_PATH=/opt/homebrew/Cellar/carta-beta/3.0.0-beta/in
==> make install
Last 15 lines from /Users/jhibbard/Library/Logs/Homebrew/carta-beta/03.make:
1 error generated.
make[2]: *** [CMakeFiles/carta_backend.dir/src/Main.cc.o] Error 1
1 error generated.
make[2]: *** [CMakeFiles/carta_backend.dir/src/DataStream/Smoothing.cc.o] Error 1
In file included from /tmp/carta-beta-20220418-83323-1avrso/src/GrpcServer/CartaGrpcService.cc:9:
In file included from /tmp/carta-beta-20220418-83323-1avrso/src/GrpcServer/CartaGrpcService.h:18:
In file included from /tmp/carta-beta-20220418-83323-1avrso/src/GrpcServer/../Session.h:52:
In file included from /tmp/carta-beta-20220418-83323-1avrso/src/Frame.h:36:
/tmp/carta-beta-20220418-83323-1avrso/src/ImageGenerators/MomentGenerator.h:12:10: fatal error: 'imageanalysis/ImageAnalysis/ImageMomentsProgressMonitor.h' file not found
#include <imageanalysis/ImageAnalysis/ImageMomentsProgressMonitor.h>
         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
1 error generated.
make[2]: *** [CMakeFiles/carta_backend.dir/src/GrpcServer/CartaGrpcService.cc.o] Error 1
make[1]: *** [CMakeFiles/carta_backend.dir/all] Error 2
make: *** [all] Error 2

If reporting this issue please do so at (not Homebrew/brew or Homebrew/core):
  https://github.com/cartavis/homebrew-tap/issues

------------------
I do have v2.0 installed (it came across when I copied the contents of my old MBP over), and it runs fine.

If it matters: 

% cmake --version
cmake version 3.23.1

CMake suite maintained and supported by Kitware (kitware.com/cmake).

jhibbard@aurelian ~ % gcc --version
Apple clang version 13.1.6 (clang-1316.0.21.2.3)
Target: arm64-apple-darwin21.4.0
Thread model: posix
InstalledDir: /Library/Developer/CommandLineTools/usr/bin

**Comment by jehibbard** [comment]
By the way, this is not urgent - I went ahead and successfully installed the electron version for the Apple M1. But I thought you might want to know.

**Comment by ajm-ska** [comment]
@jehibbard Thank you for reporting the problem. I'm sorry about my slow response. I was unaware that an issue had been opened in this repo. Thanks to @veggiesaurus for letting me know.

I figured out what was wrong. I had updated the `carta-casacore.rb` formula to use the new `3.4.0+6.5.0+2022.4.12` tag, but I forgot that I should have also updated it in the `carta.rb` and `carta-beta.rb` formulae. That is because the carta-backend needs to link to the specific carta-casacore include files when building.
```
path = HOMEBREW_PREFIX/"Cellar/carta-casacore/2022.3.9/include"

"-DCMAKE_CXX_FLAGS=-I#{path}/casacode -I#{path}/casacore",
```

So the carta-casacore files were in the `carta-casacore/2022.4.12` folder, but the carta_backend was looking in the `carta-casacore/2022.3.9` folder, and that folder did not exist on your computer.

The problem should be fixed now. All three formulae have been updated to use the even newer `2022.4.22` carta-casacore version.
